src/sim.cpp

```diff
@@ -299,6 +303,18 @@ inline void movementSystem(Engine &e,
         }
         // TODO(samk): factor out z-dimension constant and reuse when scaling cubes
     }
+    else if (type == EntityType::Vehicle && controlledState.controlledState == ControlMode::DELTA)
+    {
+        if(e.data().params.useDeltaModel)
+        {
+            forwardDeltaModel(dAction, rotation, position, velocity);
+        }
+        else
+        {
+            forwardKinematics(action, size, rotation, position, velocity);
+        }
+        // TODO(samk): factor out z-dimension constant and reuse when scaling cubes
+    }
```

Comment:
This code can be made easy to read with changing controlledState to a boolean and using BicycleModel enum 
```suggestion
    if(controlledState.controlled) 
    {
        switch (e.data().params.bicycleModel){
            case Waymax:
               //code
               
             case Delta:
               //code
        }
    }
        // TODO(samk): factor out z-dimension constant and reuse when scaling cubes
    }
```

Reply:
I have a question about it. 
- Does bicyclemodel is larger notion than waymax? I thought waymax is upper level notion, bicycle and delta are in same level notion. So that `bicycleModel` is changed to `WaymaxModel` or `DynamicsModel`.